sklearn/feature_selection/_rfe.py

```diff
@@ -156,7 +156,8 @@ def _fit(self, X, y, step_score=None):
 
         tags = self._get_tags()
         X, y = check_X_y(X, y, "csc", ensure_min_features=2,
-                         force_all_finite=not tags.get('allow_nan', True))
+                         force_all_finite=not tags.get('allow_nan', True),
+                         multi_output=True)
```

Comment:
As a side comment, I think it is indeed best to assume that multi_output is supported here, pass it to the underlying  estimator and let it error if it doesn't. As opposed to relying on `multioutput` estimator tag, which can potentially be not accurate.